packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -123,6 +124,7 @@ const TablePagination = React.forwardRef(function TablePagination(props, ref) {
             input={<InputBase className={clsx(classes.input, classes.selectRoot)} />}
             value={rowsPerPage}
             onChange={onChangeRowsPerPage}
+            id={selectId}
```

Comment:
If an id is required, should it be generated internally by Select? Meaning fixing 

https://github.com/mui-org/material-ui/blob/d49be19b669f64588d1eb0454c81b0a902826392/packages/material-ui/src/Select/SelectInput.js#L332

Reply:
Also removing the filter of #20356 as no longer useful?

Reply:
I'd like to handle this in a separate PR.